meilisearch-http/src/routes/indexes/documents.rs

```diff
@@ -138,16 +139,18 @@ pub async fn get_all_documents(
         Some(names)
     });
 
-    let documents = meilisearch
+    let (total, documents) = meilisearch
         .documents(
             path.into_inner(),
-            params.offset.unwrap_or(DEFAULT_RETRIEVE_DOCUMENTS_OFFSET),
-            params.limit.unwrap_or(DEFAULT_RETRIEVE_DOCUMENTS_LIMIT),
+            params.offset,
+            params.limit,
             attributes_to_retrieve,
         )
         .await?;
     debug!("returns: {:?}", documents);
-    Ok(HttpResponse::Ok().json(documents))
+    Ok(HttpResponse::Ok().json(json!(
+        { "limit": params.limit, "offset": params.offset, "total": total, "results": documents }
+    )))
```

Comment:
We can make a `ListView` type that id shared with all the other resources that return a list, such as listing indexes, or tasks